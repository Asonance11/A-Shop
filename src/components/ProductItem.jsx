import { React, useState } from 'react';

const ProductItem = ({ product, cart, setCart }) => {
	const [quantity, setQuantity] = useState(1);
	return (
		<div className="grid grid-cols-2 gap-8 mt-8 font-sans items-center">
			<img src={product.image} alt={product.alt} className="w-full" />
			<ul className="text-white flex flex-col gap-4 ">
				<li className="flex items-center justify-between">
					<p>Photographer:</p>
					<a href={product.photographerUrl} className="hover:underline">
						{product.photographer}
					</a>
				</li>
				<li className="flex items-center justify-between">
					<p>Price:</p>
					<p>{`$${product.price}`}</p>
				</li>
				<li className="flex items-center justify-between">
					<p>Image size</p>
					<p>{product.imageSize}</p>
				</li>
				<li className="flex items-center justify-between">
					<button
						type="button"
						className="py-2 px-4 bg-gradient-to-r from-forestGreen via-adventurousYellow to-oceanTurquoise rounded-md text-darkGrey transition-all duration-500 ease-in-out   hover:from-foggyBlue hover:to-forestGreen"
					>
						Add to Cart
					</button>
					<div className="flex items-center gap-2">
						<button
							type="button"
							className="py-1 px-2 rounded-md text-darkGrey bg-white"
							onClick={() => {
								setQuantity(quantity - 1);
							}}
							disabled={quantity <= 1 ? true : false}
						>
							-
						</button>
						<span>Quantity - {quantity}</span>
						<button
							type="button"
							className="py-1 px-2 rounded-md text-darkGrey bg-white"
							onClick={() => {
								setQuantity(quantity + 1);
							}}
						>
							+
						</button>
					</div>
				</li>
			</ul>
		</div>
	);
};

export default ProductItem;
