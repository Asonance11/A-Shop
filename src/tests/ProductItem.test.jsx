import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, it } from 'vitest';
import ProductItem from '../components/ProductItem';

describe('Product component', () => {
	it('renders properly when API call completes', async () => {
		const anItem = {
			image: 'www.awebsite.com/something.jpeg',
			alt: 'An image',
			photographer: 'me',
			price: 10000,
			imageSize: '3000 x 3000',
		};

		const { container } = render(<ProductItem product={anItem} />, {
			wrapper: MemoryRouter,
		});
		expect(container).toMatchSnapshot();
	});
});
