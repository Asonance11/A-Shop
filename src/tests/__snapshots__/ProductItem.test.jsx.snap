// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product component > renders properly when API call completes 1`] = `
<div>
  <div
    class="flex flex-col gap-8 mt-8 font-sans items-center md:grid md:grid-cols-2"
  >
    <img
      alt="An image"
      class="w-full"
      src="www.awebsite.com/something.jpeg"
    />
    <ul
      class="text-white flex flex-col gap-4 w-full"
    >
      <li
        class="flex items-center justify-between"
      >
        <p>
          Photographer:
        </p>
        <a
          class="hover:underline"
        >
          me
        </a>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <p>
          Price:
        </p>
        <p>
          $10000
        </p>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <p>
          Image size
        </p>
        <p>
          3000 x 3000
        </p>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <button
          class="py-2 px-4 bg-gradient-to-r from-forestGreen via-adventurousYellow to-oceanTurquoise rounded-md text-darkGrey transition-all duration-500 ease-in-out   hover:from-foggyBlue hover:to-forestGreen"
          type="button"
        >
          Add to Cart
        </button>
        <div
          class="flex items-center gap-2"
        >
          <button
            class="py-1 px-2 rounded-md text-darkGrey bg-white"
            disabled=""
            type="button"
          >
            -
          </button>
          <span>
            Quantity - 
            1
          </span>
          <button
            class="py-1 px-2 rounded-md text-darkGrey bg-white"
            type="button"
          >
            +
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
